hack/util.sh

```diff
@@ -660,7 +660,19 @@ function install_registry() {
 readonly -f install_registry
 
 function wait_for_registry() {
-	wait_for_command "oc get endpoints docker-registry --template='{{ len .subsets }}' --config='${ADMIN_KUBECONFIG}' | grep -q '[1-9][0-9]*'" $((5*TIME_MIN))
+	local latestversion=$(oc get dc/docker-registry -o jsonpath='{.status.latestVersion}' --config="${ADMIN_KUBECONFIG}")
+	if [[ -z "${latestversion:-}" ]]; then
+		echo "Failed to get latest version of docker-registry deployment config"
+		exit 1
+	fi
+	wait_for_command "oc get rc/docker-registry-$latestversion --template \"{{ index .metadata.annotations \\\"openshift.io/deployment.phase\\\" }}\" --config='${ADMIN_KUBECONFIG}' | grep Complete"  $((5*TIME_MIN))
+	local registrypod=$(oc get pod -l "deployment=docker-registry-$latestversion" -o jsonpath=$'{.items[*].metadata.name}\n' --config="${ADMIN_KUBECONFIG}" | head -n 1)
+	if [[ -z "${registrypod:-}" ]]; then
+		echo "Failed to get docker registry pod name"
+		exit 1
+	fi
+	local podnamejs='{range .subsets[*]}{range .addresses[*]}{.targetRef.name},{end}{end}'
+	wait_for_command "oc get endpoints/docker-registry -o 'jsonpath=$podnamejs' --config='${ADMIN_KUBECONFIG}' | egrep -q '(^|,)${registrypod},'" $TIME_MIN
```

Comment:
Here's a log from one of the recent runs:

```
[INFO] Waiting for command to finish: 'oc get rc/docker-registry-2 --template "{{ index .metadata.annotations \"openshift.io/deployment.phase\" }}" --config='/tmp/openshift/test-end-to-end-docker//openshift.local.config/master/admin.kubeconfig' | grep Complete'...
Complete
[INFO] Success running command: 'oc get rc/docker-registry-2 --template "{{ index .metadata.annotations \"openshift.io/deployment.phase\" }}" --config='/tmp/openshift/test-end-to-end-docker//openshift.local.config/master/admin.kubeconfig' | grep Complete' after 16 seconds
[INFO] Waiting for command to finish: 'oc get endpoints/docker-registry -o 'jsonpath={range .subsets[*]}{range .addresses[*]}{.targetRef.name},{end}{end}' --config='/tmp/openshift/test-end-to-end-docker//openshift.local.config/master/admin.kubeconfig' | egrep -q '(^|,)docker-registry-2-n8o46,''...
[INFO] Success running command: 'oc get endpoints/docker-registry -o 'jsonpath={range .subsets[*]}{range .addresses[*]}{.targetRef.name},{end}{end}' --config='/tmp/openshift/test-end-to-end-docker//openshift.local.config/master/admin.kubeconfig' | egrep -q '(^|,)docker-registry-2-n8o46,'' after 1 seconds
[INFO] Waiting for command to finish: 'oc get pod -l deploymentconfig=docker-registry -o jsonpath='{.items[*].status.conditions[?(@.type=="Ready")].status}' --config='/tmp/openshift/test-end-to-end-docker//openshift.local.config/master/admin.kubeconfig' | grep -qi true'...
[INFO] Success running command: 'oc get pod -l deploymentconfig=docker-registry -o jsonpath='{.items[*].status.conditions[?(@.type=="Ready")].status}' --config='/tmp/openshift/test-end-to-end-docker//openshift.local.config/master/admin.kubeconfig' | grep -qi true' after 0 seconds
```

The `oc get endpoint` needed additional 1 second to succeed after the deployment of dc-2 completed which IMHO justifies first two wait statements. I'm still not sure about the third (wait for a pod to become ready). I'd prefer to keep it though.